Add more docs on durable state (#30403)
akka/akka: akka-docs/src/main/paradox/typed/persistence-durable-state.md

The current state is always stored in the database. Since only the latest state is stored, we don't have access to any of the history of changes, unlike event sourced storage. Akka Persistence would read that state and store it in memory. After processing of the command is finished, the new state will be stored in the database. The processing of the next command will not start until the state has been successfully stored in the database.

> Akka Persistence also supports @ref:[Event Sourcing](persistence.md) based implementation, where only the _events_ that 
> are persisted by the actor are stored, not the actual state of the actor. By storing all events, using this model, 
> a stateful actor can be recovered by replaying the stored events to the actor, allowing it to rebuild its state.
> 

Reviewer: ```suggestion
Akka Persistence also supports @ref:[Event Sourcing](persistence.md) based implementation, where only the _events_ that are persisted by the actor are stored, but. not the actual state of the actor. By storing all events, using this model, 
a stateful actor can be recovered by replaying the stored events to the actor, which allows it to rebuild its state.

```

Author: done ..

The current state is always stored in the database. Since only the latest state is stored, we don't have access to any of the history of changes, unlike event sourced storage. Akka Persistence would read that state and store it in memory. After processing of the command is finished, the new state will be stored in the database. The processing of the next command will not start until the state has been successfully stored in the database.

> Akka Persistence also supports @ref:[Event Sourcing](persistence.md) based implementation, where only the _events_ that are persisted by the actor are stored, but. not the actual state of the actor. By storing all events, using this model, 

Reviewer 2: ```suggestion
Akka Persistence also supports @ref:[Event Sourcing](persistence.md) based implementation, where only the _events_ that are persisted by the actor are stored, not the actual state of the actor. By storing all events, using this model, 
```

The two most commonly used effects are: 

> * `persist` will persist the latest value of the state. No history of state changes will be stored though

Reviewer 2: ```suggestion
* `persist` will persist the latest value of the state. No history of state changes will be stored
```
